### PR TITLE
add sfml/2.5.1

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

Taken from https://github.com/bincrafters/community/tree/main/recipes/sfml, with several improvements:

- Macos M1 support
- do not force window if graphics enabled, raise instead if window is disabled
- allow to buid with shared & MT/MTd runtime
- fix iOS detection (and properly propagate OpenGLES, but I think that it should come from `opengl` recipe).
- proper CMake imported targets (not namespaced, instead they have a `sfml-` prefix)
- properly emulate official `sfml-all.pc` pkgconfig file
- robust discovery/injection of dependencies
- ogg removed from direct dependencies (it's just a transitive dependency through vorbis)

Linux not supported yet (it requires `libudev`).

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by szavadsky — DISMISSED** [review]

File: recipes/sfml/all/patches/0002-allow-non-x86-64-macos.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++ b/CMakeLists.txt
+@@ -253,7 +253,7 @@ if(SFML_OS_MACOSX)
+     endif()
+ 
+     # only the default architecture (i.e. 64-bit) is supported
+-    if(NOT CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
++    if(0)
+         message(FATAL_ERROR "Only 64-bit architecture is supported")
+     endif()
+
```
Comment (intelligide) [inline recipes/sfml/all/patches/0002-allow-non-x86-64-macos.patch:13]: I don't think it's wise to enable a platform that is not officially supported. Does the library work on non x86-64 macos ?

File: recipes/sfml/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+            {values["target"]: "SFML::{}".format(component) for component, values in self._sfml_components.items()}
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
```
Comment (blackliner) [inline recipes/sfml/all/conanfile.py:120]: Why not create an ALIAS library?

File: recipes/sfml/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
```
Comment (blackliner) [inline recipes/sfml/all/test_package/CMakeLists.txt:7]: Modern cmake, dont use ${PROJECT_NAME}, write trst_package instead

Comment (SpaceIm) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: ?

File: recipes/sfml/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/sfml/all/conanfile.py:120]: because it can't work

File: recipes/sfml/all/patches/0002-allow-non-x86-64-macos.patch (hunk 1 above)
Comment (SpaceIm) [reply recipes/sfml/all/patches/0002-allow-non-x86-64-macos.patch:13]: It compiles and link, maybe someone with macOS M1 can confirm that SFML properly works.

**Review by madebr — DISMISSED** [review]

**Review by intelligide — DISMISSED** [review]

**Comment by SpaceIm** [comment]
Some Android folks who can tell how OpenGLES should be link?

**Comment by madebr** [comment]
@SpaceIm 
When I tried building sdl for android, I had to add `GLESv1_CM` and `GLESv2` to the system libs.

File: recipes/sfml/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (blackliner) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: Here you go: https://www.reddit.com/r/cmake/comments/j4xysx/modern_cmake_tips/g7vl9nj/

Comment (SpaceIm) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: Honestly, in test_package it doesn't matter, and it has nothing to do with "Modern CMake".

Comment (blackliner) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: Being a good example is the cause here, but it's your call 🤙

Comment (SpaceIm) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: It's fine to use `PROJECT_NAME` with only one target, not sure that a comment on reddit is relevant ;)

**Comment by SpaceIm** [comment]
> @SpaceIm
> When I tried building sdl for android, I had to add `GLESv1_CM` and `GLESv2` to the system libs.

SFML seems to link to EGL & GLESv1_CM: https://github.com/SFML/SFML/blob/2f11710abc5aa478503a7ff3f9e654bd2078ebab/src/SFML/Window/CMakeLists.txt#L253-L255

File: recipes/sfml/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (prince-chrismc) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: > Modern cmake, dont use ${PROJECT_NAME}, write test_package instead

That's the first time I've ever heard that one 😱 there's a few of top notch OSS, like curl, that do this so it's hard to say its a best practice.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/sfml/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (blackliner) [reply recipes/sfml/all/test_package/CMakeLists.txt:7]: yea, and curl != modern ;-)

**Comment by SpaceIm** [comment]
No progress in CI (last log at 13.02 UTC)? It takes 5 minutes in github action to build all these configurations. Could it be boost PR which steal all the agents?

**Review by madebr — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**